sql-statements/sql-statement-create-index.md

```diff
@@ -232,6 +232,120 @@ Therefore, when the query performance outweighs the insert and update performanc
+Query OK, 1 row affected (0.01 sec)
+
+mysql> INSERT INTO customers VALUES (1, 'pingcap', '{"zipcode": [2,3]}');
+ERROR 1062 (23000): Duplicate entry '2' for key 'customers.zips'
+```
+
+The same record can have duplicate values, but when different records have duplicate values, an error is reported.
+
+```sql
+-- Insert successfully
+mysql> INSERT INTO t1 VALUES('[1,1,2]');
+mysql> INSERT INTO t1 VALUES('[3,3,3,4,4,4]');
+
+-- Insert failed
+mysql> INSERT INTO t1 VALUES('[1,2]');
+mysql> INSERT INTO t1 VALUES('[2,3]');
+```
+
+You can also define a multi-valued index as a composite index:
+
+```sql
+mysql> CREATE TABLE customers (
+    id BIGINT NOT NULL AUTO_INCREMENT PRIMARY KEY,
+    name CHAR(10),
+    custinfo JSON,
+    INDEX zips(name, (CAST(custinfo->'$.zipcode' AS UNSIGNED ARRAY)))
+);
+```
+
+When a multi-valued index is defined as a composite index, the multi-valued part can appear in any position, but only once.
+
+```sql
+mysql> CREATE TABLE customers (
+    id BIGINT NOT NULL AUTO_INCREMENT PRIMARY KEY,
+    name CHAR(10),
+    custinfo JSON,
+    INDEX zips(name, (CAST(custinfo->'$.zipcode' AS UNSIGNED ARRAY)), (CAST(custinfo->'$.zipcode' AS UNSIGNED ARRAY)))
+);
+ERROR 1235 (42000): This version of TiDB doesn't yet support 'more than one multi-valued key part per index'.
+```
+
+The written data must exactly match the type defined by the multi-valued index; otherwise, the data write fails:
+
+```sql
+-- All elements in the zipcode field must be the UNSIGNED type.
+mysql> INSERT INTO customers VALUES (1, 'pingcap', '{"zipcode": [-1]}');
+ERROR 3752 (HY000): Value is out of range for expression index 'zips' at row 1
+
+mysql> INSERT INTO customers VALUES (1, 'pingcap', '{"zipcode": ["1"]}'); -- Incompatible with MySQL
+ERROR 3903 (HY000): Invalid JSON value for CAST for expression index 'zips'
+
+mysql> INSERT INTO customers VALUES (1, 'pingcap', '{"zipcode": [1]}');
+Query OK, 1 row affected (0.00 sec)
+```
+
+### Use a multi-valued index
+
+See [Index Selection - Use multi-valued indexes](/choose-index.md#use-a-multi-valued-index) for more details.
+
+### Limitations
+
+- For an empty JSON array, no corresponding index record is generated.
+- The target type in `CAST(... AS ... ARRAY)` cannot be any of `BINARY`, `JSON`, `YEAR`, `FLOAT`, `DOUBLE`, and `DECIMAL`. The source type must be JSON.
+- You cannot use a multi-valued index for sorting.
+- You can only create a multi-valued index on a JSON array.
+- A multi-valued index cannot be a primary key or a foreign key.
+- The extra storage space used by a multi-valued index = the average number of array elements per row * the space used by a normal secondary index.
+- Compared with normal indexes, DML operations will modify more index records for multi-valued indexes, so multi-valued indexes will have a greater performance impact than normal indexes.
+- Because multi-valued indexes are a special type of expression index, they are the limitations of expression indexes.
```

Comment:
```suggestion
- Because multi-valued indexes are a special type of expression index, multi-valued indexes have the same limitations as expression indexes.
```